### PR TITLE
Feature: Identify Subscribers in Sentry

### PR DESCRIPTION
**Pull request by jakeleboeuf**
# Describe your PR
To support future debugging efforts, we want to `identify` subscribed users in Sentry.

## Steps to test

1. `$ npm run test npm run test src/utils/globalId.test.js`
2. Look for ✅ 

### Additional notes
To comply with GDPR, we're creating a non-identifying _Global Id_ for the user to store in Sentry. This will require setting a new secret `RBB_SECRET` in our prod env. It can be anything :)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,14 +27,15 @@
...
+    "base-64": "^0.1.0",
     "dotenv": "^8.2.0",
     "emotion-theming": "^10.0.27",
     "gatsby": "^2.23.3",
     "gatsby-cypress-endpoints": "0.0.6",
     "gatsby-plugin-chakra-ui": "^0.1.4",
     "gatsby-plugin-create-client-paths": "^2.3.4",
     "gatsby-plugin-facebook-pixel": "^1.0.3",
-    "gatsby-plugin-google-analytics": "^2.3.4",
+    "gatsby-plugin-google-tagmanager": "^2.3.4",
```
Comment (jakeleboeuf) [inline package.json:38]: Not sure where this slipped in from...

File: src/components/SubscribeForm.js [hunk 2]
```diff
@@ -1,28 +1,69 @@
...
+import * as Sentry from '@sentry/browser';
 
 import { FormControl, FormLabel, Input, Stack, Text } from '@chakra-ui/core';
 import Button from './Button';
+import { toGlobalId } from '../utils/globalId';
+
+// Some configurable things
+const FORM_USER = 'a5fccdceb916e00a46d603ed8';
+const FORM_ID = '992a06eec8';
+export const FORM_ACTION = `https://rebuildblackbusiness.us10.list-manage.com/subscribe/post?u=${FORM_USER}&amp;id=${FORM_ID}`;
```
Comment (jakeleboeuf) [inline src/components/SubscribeForm.js:11]: Mad a few of these keys a bit more reusable

File: src/components/SubscribeForm.js [hunk 3]
```diff
@@ -1,28 +1,69 @@
...
 
 const SubscribeForm = () => {
-  const formAction =
-    'https://rebuildblackbusiness.us10.list-manage.com/subscribe/post?u=a5fccdceb916e00a46d603ed8&amp;id=992a06eec8';
+  const [email, setEmail] = React.useState('');
+
+  const identifySentryUser = () => {
+    const id = toGlobalId({ type: FORM_ID, id: email });
```
Comment (jakeleboeuf) [inline src/components/SubscribeForm.js:17]: Creating a Global ID (UUID) to keep user info out of Sentry

File: package.json (hunk 1 above)
Comment (racedale) [reply package.json:38]: It'd be from this, but it's going backwards
https://github.com/Rebuild-Black-Business/RBB-Website/pull/198

File: src/utils/globalId.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import base64 from 'base-64';
+
+export const toGlobalId = ({ type, id }) =>
+  base64.encode(`${type}:${id}:${process.env.RBB_SECRET}`);
```
Comment (jakeleboeuf) [inline src/utils/globalId.js:4]: @racedale Note we need to set some secret in prod- again, it can be whatever you like

**Comment by magnificode** [comment]
Going to defer this review to @racedale - our resident Sentry wizard :D

File: src/components/SubscribeForm.js (hunk 2 above)
Comment (racedale) [reply src/components/SubscribeForm.js:11]: Love the documentation for reasons of change!

**Pull request merged**